geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterConfigurationService.java

```diff
@@ -319,7 +319,13 @@ public boolean addJarsToThisLocator(String[] jarNames, byte[][] jarBytes, String
         // will need the jars on file to upload to other locators. Need to update the jars
         // using a new copy of the Configuration so that the change listener will pick up the jar
         // name changes.
+        // The process happens in 2 phases to cater for the situation where an existing jar is only
+        // updated.
+
         Configuration configurationCopy = new Configuration(configuration);
+        configurationCopy.removeJarNames(jarNames);
```

Comment:
we don't need to this in 2 phases now, just the add since we are getting all the existing jar files from locators anyway.